tests/unit_tests/db_engine_specs/test_databend.py

```diff
@@ -0,0 +1,145 @@
+
+    cursor = Mock()
+    cursor.execute.side_effect = NewConnectionError(
+        "Dummypool", "Exception with sensitive data"
+    )
+    with pytest.raises(SupersetDBAPIDatabaseError) as ex:
+        DatabendEngineSpec.execute(cursor, "SELECT col1 from table1")
+
+
+@pytest.mark.parametrize(
+    "target_type,expected_result",
+    [
+        ("Date", "to_date('2019-01-02')"),
+        ("DateTime", "to_dateTime('2019-01-02 03:04:05')"),
+        ("UnknownType", None),
+    ],
+)
+def test_connect_convert_dttm(
+    target_type: str, expected_result: Optional[str], dttm: datetime
+) -> None:
+    from superset.db_engine_specs.databend import DatabendEngineSpec as spec
+
+    assert_convert_dttm(spec, target_type, expected_result, dttm)
+
+
+@pytest.mark.parametrize(
+    "native_type,sqla_type,attrs,generic_type,is_dttm",
+    [
+        ("Varchar", String, None, GenericDataType.STRING, False),
+        ("Nullable(Varchar)", String, None, GenericDataType.STRING, False),
+        ("Array(UInt8)", String, None, GenericDataType.STRING, False),
+        ("Int8", Integer, None, GenericDataType.NUMERIC, False),
+        ("Int16", Integer, None, GenericDataType.NUMERIC, False),
+        ("Int32", Integer, None, GenericDataType.NUMERIC, False),
+        ("Int64", Integer, None, GenericDataType.NUMERIC, False),
+        ("Int128", Integer, None, GenericDataType.NUMERIC, False),
+        ("Int256", Integer, None, GenericDataType.NUMERIC, False),
+        ("Nullable(Int64)", Integer, None, GenericDataType.NUMERIC, False),
+        ("UInt8", Integer, None, GenericDataType.NUMERIC, False),
+        ("UInt16", Integer, None, GenericDataType.NUMERIC, False),
+        ("UInt32", Integer, None, GenericDataType.NUMERIC, False),
+        ("UInt64", Integer, None, GenericDataType.NUMERIC, False),
+        ("UInt128", Integer, None, GenericDataType.NUMERIC, False),
+        ("UInt256", Integer, None, GenericDataType.NUMERIC, False),
+        ("Float", Float, None, GenericDataType.NUMERIC, False),
+        ("Double", Float, None, GenericDataType.NUMERIC, False),
+        ("Decimal(1, 2)", DECIMAL, None, GenericDataType.NUMERIC, False),
+        ("Decimal32(2)", DECIMAL, None, GenericDataType.NUMERIC, False),
+        ("Decimal64(2)", DECIMAL, None, GenericDataType.NUMERIC, False),
+        ("Decimal128(2)", DECIMAL, None, GenericDataType.NUMERIC, False),
+        ("Decimal256(2)", DECIMAL, None, GenericDataType.NUMERIC, False),
+        ("Bool", Boolean, None, GenericDataType.BOOLEAN, False),
+        ("Nullable(Bool)", Boolean, None, GenericDataType.BOOLEAN, False),
+        ("Date", Date, None, GenericDataType.TEMPORAL, True),
+        ("Nullable(Date)", Date, None, GenericDataType.TEMPORAL, True),
+        ("Datetime", DateTime, None, GenericDataType.TEMPORAL, True),
+        ("Nullable(Datetime)", DateTime, None, GenericDataType.TEMPORAL, True),
+    ],
+)
+def test_connect_get_column_spec(
```

Comment:
Not that it matters that much, but just curious why this has `connect` in the function name?